types/json-patch-gen/index.d.ts

```diff
@@ -0,0 +1,21 @@
+// Type definitions for json-patch-gen 1.0
+// Project: https://github.com/gregsexton/json-patch-gen
+// Definitions by: Konstantin Rohde <https://github.com/RohdeK>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.2
+
+type PatchOperation = "replace" | "add" | "remove";
+
+interface JsonPatch {
+    op: PatchOperation;
+    path: string;
+    value: any;
+}
+
+type Diff = (obj1: object, obj2: object) => JsonPatch[];
```

Comment:
confusingly, the *type* `object` doesn't include `null`, even though `typeof null === "object"` at runtime and [`diff` allows `obj1` and `obj2` to be null](https://github.com/gregsexton/json-patch-gen/blob/86a801266c7f9d9505de9ebe76c309251710adde/lib/diff.js#L133). The types of obj1 and obj2 should be `object | null`.